Assets/Scripts/VFX/RenderSettings/RenderSettingsManager.cs

```diff
@@ -0,0 +1,193 @@
+﻿using UnityEngine;
+using System.Collections;
+
+[ExecuteInEditMode]
+public class RenderSettingsManager : SingletonBehaviour<RenderSettingsManager> 
+{
+	LightingSettings _currentLightingSettings = new LightingSettings();
+
+	[ReadOnly, SerializeField] 
+	TimeLightingSettings _currentTimeRenderSettings = new TimeLightingSettings();
+
+	Texture2D _gradientTexture = null;
+	public Texture2D gradientTexture
+	{
+		get
+		{
+			if( !_gradientTexture )
+			{
+				_gradientTexture = new Texture2D( 128, 1, TextureFormat.ARGB32, false );
+			}
+
+			return _gradientTexture;
+		}
+	}
+
+	[SerializeField] Light _directionalLight = null;
+	int _skyboxTintPropertyID = 0;
+
+	float _skyboxInitRotation = 0f;
+	[SerializeField, Range(0f, 5f)] float _skyboxRotSpeed = 1f;
+
+	[Tooltip( "The percent of the end of the day cycle that stays completely dark." )]
+	[SerializeField] float _nightFraction = 0.1f;
+
+	UnityStandardAssets.ImageEffects.GlobalFog _globalFog = null;
+	public UnityStandardAssets.ImageEffects.GlobalFog globalFog
+	{
+		get
+		{
+			if( !_globalFog )
+			{
+				_globalFog = GameObject.FindObjectOfType<UnityStandardAssets.ImageEffects.GlobalFog>();
+				if( !_globalFog )
+				{
+					Debug.LogWarning( "Global fog doesn't exist so render settings will be broken" );
+				}
+			}
+
+			return _globalFog;
+		}
+	}
+
+	[ReadOnly("Daylight Intensity"), Tooltip( "0 is midnight. 1 is noon." ), SerializeField] 
+	float _daylightIntensity = 1.0f;
+	public static float daylightIntensity
+	{
+		get { return instance._daylightIntensity; }
+		private set { instance._daylightIntensity = value; }
+	}
+
+	void Awake()
+	{
+		_skyboxTintPropertyID = Shader.PropertyToID( "_Tint" );
+		RenderSettings.skybox.SetFloat( "_Rotation", _skyboxInitRotation );
+
+		_gradientTexture = null;
+		_gradientTexture = new Texture2D( 128, 1, TextureFormat.ARGB32, false );
+
+		UpdateTimeOfDay();
+		UpdateFogGradientTexture();
+	}
+
+	void Update()
+	{
+		RenderSettings.skybox.SetFloat( "_Rotation", Time.realtimeSinceStartup * _skyboxRotSpeed );
+	
+		UpdateRenderSettings();
+	}
+
+	void UpdateTimeOfDay()
+	{
+		float timeOfDay = DayCycleManager.currentTime / ( DayCycleManager.dayCycleLength * ( 1.0f - _nightFraction) );
+		timeOfDay = Mathf.Clamp01( timeOfDay );
+		
+		daylightIntensity = Mathf.Cos( timeOfDay * 2.0f * Mathf.PI ) * -0.5f + 0.5f;
+		
+		TimeLightingSettings.Lerp( _currentLightingSettings.nightSettings,
+		                           _currentLightingSettings.daySettings,
+		                           daylightIntensity,
+		                           ref _currentTimeRenderSettings );
```

Comment:
i would pull this out and put it in the update method because UpdateTimeOfDay() doesn't sound like it should set the current render settings.